src/applications/vaos/utils/selectors.js

```diff
@@ -544,45 +544,74 @@ export function selectActiveExpressCareFacility(state, nowMoment) {
+}
+
+/**
+ * Returns next schedulable window.  If today is schedulable and current time is before window,
+ * return today's window.  Otherwise, return the next schedulable day's window
  */
-export function selectExpressCareHours(state) {
+export function selectNextAvailableExpressCareWindowString(state, nowMoment) {
   if (!state.expressCare.supportedFacilities?.length) {
     return null;
   }
 
   const facility = state.expressCare.supportedFacilities[0];
   const siteId = facility.facilityId.substring(0, 3);
+  const { timezone } = getTimezoneBySystemId(siteId);
   const timezoneAbbreviation = getTimezoneAbbrBySystemId(siteId);
-
-  // Trying to group days with the same times together
-  const windows = new Map();
-  facility.days.forEach(day => {
-    const key = `${day.startTime}_${day.endTime}`;
-    if (windows.has(key)) {
-      windows.get(key).push(day.day);
+  const nowFacilityTime = nowMoment.clone().tz(timezone);
+  const dayOfWeek = nowFacilityTime.format('dddd').toUpperCase();
+  const todayDayOfWeekIndex = Number(nowFacilityTime.format('d'));
+  const todaysWindow = facility.days.find(d => d.day === dayOfWeek);
+
+  // Sort schedulable days starting from today so we can easily find the next
+  // day if needed
+  const daysSortedStartingToday = [
+    todaysWindow,
+    ...facility.days.filter(day => day.dayOfWeekIndex > todayDayOfWeekIndex),
+    ...facility.days.filter(day => day.dayOfWeekIndex < todayDayOfWeekIndex),
+  ].filter(day => !!day);
+
+  if (todaysWindow) {
+    const start = moment.tz(
+      `${nowFacilityTime.format('YYYY-MM-DD')}T${todaysWindow.startTime}:00`,
+      timezone,
+    );
+    if (nowFacilityTime.isBefore(start)) {
+      // If today is schedulable and we are before the window, return today's window
+      return getWindowString(todaysWindow, timezoneAbbreviation, true);
     } else {
-      windows.set(key, [day.day]);
+      // In the rare case the today is the only schedulable day and they are past the window,
+      // return today's window and specify "next week". Otherwise, return the next schedulable day
+      if (daysSortedStartingToday.length === 1) {
+        return `next ${getWindowString(
+          todaysWindow,
+          timezoneAbbreviation,
+          false,
+        )}`;
+      }
+      return getWindowString(
+        daysSortedStartingToday[1],
+        timezoneAbbreviation,
+        false,
+      );
     }
-  });
-
-  const segments = Array.from(windows.entries());
-
-  // Formatting each segment as 'day, day from x to y'
-  return joinWithAnd(
-    segments.map(([hours, days]) => {
-      const [start, end] = hours.split('_');
-      return `${days.map(titleCase).join(', ')} from ${moment(
-        start,
-        'HH:mm',
-      ).format('h:mm a')} to ${moment(end, 'HH:mm').format(
-        'h:mm a',
-      )} ${timezoneAbbreviation}`;
-    }),
-  );
+  } else {
+    // If today isn't schedulable, return the next day that is
+    return getWindowString(
```

Comment:
Surprised this isn't covered in our tests. We should make sure this branch is covered.